### ISSUE TITLE
Open source blocks implementations

### ISSUE THREAD

**Issue by hammer** [refstudio/refstudio#9]
We should do a more thorough evaluation of OSS libraries that could provide a Notion blocks-like writing experience.

- https://github.com/TypeCellOS/BlockNote
- https://github.com/AppFlowy-IO/AppFlowy: could possibly refactor their blocks implementation out?

**Comment by danvk** [comment]
I filed an issue about block folding on BlockNote: https://github.com/TypeCellOS/BlockNote/issues/196

AppFlowy is written in Dart / Flutter, which is probably not the stack we want to use. I also did not see any evidence that it supported folding.

**Comment by hammer** [comment]
Thanks @danvk! I did a bit more digging and turned up https://github.com/liveblocks/liveblocks, which is made by a venture-backed company so I'm not quite sure what is OSS and what is in their hosted offering. I searched the repo for references to ProseMirror, TipTap, and Lexical and did not find any.

I also searched on the Lexical repo for evidence of people trying to build Notion-like features. A slash-invoked menu selector was pushed in https://github.com/facebook/lexical/pull/2534 and is known as the `ComponentPickerPlugin`, and https://github.com/facebook/lexical/pull/2860 pushed a `DraggableBlockPlugin`.  In various places the developers have indicated they don't see blocks as something that will ever be a first-class object in Lexical.
